### ISSUE TITLE
Backpropagation is extremely slow (with 155s)

### ISSUE THREAD

**Issue by sookiemm** [MzeroMiko/VMamba#333]
I used the VSSM module, but the backpropagation is very slow, I want to know why this is caused? And what was the backpropagation time in the authors' experiments? What acceleration strategy did the author use?

Here is my code for training:
`        for batch in train_dl:
            optimizer.zero_grad()
            batch = to_cuda(batch)
            images = batch['image']
            img_size = images.size()[2:]
            start_time = time.time()
            with torch.autocast(device_type='cuda', dtype=torch.float16, enabled=fp16):
                outputs = model(images)
                loss_dict = criterion(outputs, batch)

            inference_time = time.time() - start_time  # Time taken for inference
            print(f"Batch Inference time: {inference_time:.4f}s")
            # Log loss values
              loss_value = loss_dict.detach().item()
              batch_size = outputs.size(0)
              train_loss.update(loss_value / batch_size, batch_size)
            # Measure backward pass time
            start_time = time.time()
            scaler.scale(loss_dict['total']).backward()
            scaler.step(optimizer)
            scaler.update()
            backward_time = time.time() - start_time  # Time taken for backward pass
            print(f"Batch Backward time: {backward_time:.4f}s")`                                                                                                                                              I used the same code to experiment with VSSM+decoder+head and Swin+decoder+head at the same depths and got the following results.                                                                               `vmamba:
Batch Inference time: 15.1769s
Batch Backward time: 155.6933s
Swin:
Batch Inference time: 6.3927s
Batch Backward time: 0.3513s
`

**Comment by sookiemm** [comment]
I solved this problems referring to this answer:
https://github.com/MzeroMiko/VMamba/issues/117#issuecomment-2106254025
I also encountered many version dependence problems when installing the packages. And I sovled them with: 
https://github.com/MzeroMiko/VMamba/issues/117#issuecomment-2106254025

Now the reference and backpropagation time return to normal
Batch Inference time: 6.6336s
Batch Backward time: 1.5650s
Batch Inference time: 0.0684s
Batch Backward time: 0.0835s
Batch Inference time: 0.0374s
Batch Backward time: 0.0655s

**Comment by MzeroMiko** [comment]
For the environment, here's a take-away solution (torch >= 2.1 is ok):

```python
    conda create -n ${NAME} python==3.10 -y
    which conda && which python && which pip
    pip install torch==2.2 torchvision torchaudio triton pytest chardet yacs termcolor fvcore seaborn packaging ninja einops numpy==1.24.4 timm==0.4.12
    pip install tensorboard
    CUDA_HOME=/usr/local/cuda-12.1/ pip install kernels/selective_scan
```
